### PR TITLE
Remove entry and allocated memory for empty buffers.

### PR DESCRIPTION
**Pull request by tryangul**
## What
Fixes memory leak due to empty buffers holding onto allocated memory. Fix for [this](https://github.com/airbytehq/oncall/issues/3168) issue.

## How
If a buffer for a stream descriptor ever empties, we shrink the queue and release any lingering memory not covered by the pending batch release.

## Notes
We are shrinking the queue and not removing it because 1) enough code assumed the queue was always present that it presented a bigger change. 2) if there is some race on the map (there shouldn't be) removing the entry could lead to dropped data.

## 🚨 User Impact 🚨
This fixes an ongoing issue. If users use an async connector on tables with lots of streams, they will want to upgrade.

### CODE REVIEW

File: airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/buffers/BufferDequeue.java [hunk 1]
```diff
@@ -74,7 +74,14 @@ public MemoryAwareMessageBatch take(final StreamDescriptor streamDescriptor, fin
         }
       }
 
-      queue.addMaxMemory(-bytesRead.get());
+      if (queue.isEmpty()) {
+        buffers.remove(streamDescriptor);
+        // free the remainder, leaving the read bytes to be freed by the batch flush
+        final var slop = queue.getMaxMemoryUsage() - bytesRead.get();
```
Comment (benmoriceau) [inline airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/buffers/BufferDequeue.java:80]: Is this right? I feel that the maxMemory usage is a fix number.

Comment (tryangul) [reply airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/buffers/BufferDequeue.java:80]: We add to maxMemory as we allocate [here](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/core/src/main/java/io/airbyte/cdk/integrations/destination_async/buffers/BufferEnqueue.java#L60). From what I can tell `maxMemory` represents the total memory allocated for the buffer

**Comment by tryangul** [comment]
@benmoriceau should be ready for re-review

**Comment by airbyte-oss-build-runner** [comment]
##  destination-redshift test report (commit [`480a1a91eb`](https://github.com/airbytehq/airbyte/pull/31194/commits/480a1a91eb71e3c1c7af55b05c1a3d8bd94fcd2c)) - ❌

⏲️  Total pipeline duration: 19mn51s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Build connector tar                                                  | ✅        |
| Build destination-redshift docker image for platform(s) linux/x86_64 | ✅        |
| Java Connector Unit Tests                                            | ✅        |
| Build airbyte/normalization-redshift:dev                             | ✅        |
| Java Connector Integration Tests                                     | ❌        |
| Validate metadata for destination-redshift                           | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1696987058/480a1a91eb71e3c1c7af55b05c1a3d8bd94fcd2c/destination-redshift/0.6.9/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:480a1a91eb71e3c1c7af55b05c1a3d8bd94fcd2c)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-redshift test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`deca106787`](https://github.com/airbytehq/airbyte/pull/31194/commits/deca106787b1a4505ef04a442f0ed7873934bc95)) - ❌

⏲️  Total pipeline duration: 01mn48s 

| Step                                       | Result   |
|:-------------------------------------------|:---------|
| Build connector tar                        | ❌        |
| Validate metadata for destination-bigquery | ✅        |
| Connector version semver check             | ✅        |
| Connector version increment check          | ✅        |
| QA checks                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1696991624/deca106787b1a4505ef04a442f0ed7873934bc95/destination-bigquery/2.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:deca106787b1a4505ef04a442f0ed7873934bc95)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-snowflake test report (commit [`deca106787`](https://github.com/airbytehq/airbyte/pull/31194/commits/deca106787b1a4505ef04a442f0ed7873934bc95)) - ❌

⏲️  Total pipeline duration: 01mn09s 

| Step                                        | Result   |
|:--------------------------------------------|:---------|
| Build connector tar                         | ❌        |
| Validate metadata for destination-snowflake | ✅        |
| Connector version semver check              | ✅        |
| Connector version increment check           | ✅        |
| QA checks                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1696991624/deca106787b1a4505ef04a442f0ed7873934bc95/destination-snowflake/3.2.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:deca106787b1a4505ef04a442f0ed7873934bc95)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-redshift test report (commit [`deca106787`](https://github.com/airbytehq/airbyte/pull/31194/commits/deca106787b1a4505ef04a442f0ed7873934bc95)) - ❌

⏲️  Total pipeline duration: 01mn14s 

| Step                                       | Result   |
|:-------------------------------------------|:---------|
| Build connector tar                        | ❌        |
| Validate metadata for destination-redshift | ✅        |
| Connector version semver check             | ✅        |
| QA checks                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1696991624/deca106787b1a4505ef04a442f0ed7873934bc95/destination-redshift/0.6.10/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:deca106787b1a4505ef04a442f0ed7873934bc95)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-redshift test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`164a81877b`](https://github.com/airbytehq/airbyte/pull/31194/commits/164a81877b096e4df72869861310e909b2103d32)) - ❌

⏲️  Total pipeline duration: 01mn41s 

| Step                                       | Result   |
|:-------------------------------------------|:---------|
| Build connector tar                        | ❌        |
| Validate metadata for destination-bigquery | ✅        |
| Connector version semver check             | ✅        |
| Connector version increment check          | ✅        |
| QA checks                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1696992528/164a81877b096e4df72869861310e909b2103d32/destination-bigquery/2.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:164a81877b096e4df72869861310e909b2103d32)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-redshift test report (commit [`164a81877b`](https://github.com/airbytehq/airbyte/pull/31194/commits/164a81877b096e4df72869861310e909b2103d32)) - ❌

⏲️  Total pipeline duration: 01mn14s 

| Step                                       | Result   |
|:-------------------------------------------|:---------|
| Build connector tar                        | ❌        |
| Validate metadata for destination-redshift | ✅        |
| Connector version semver check             | ✅        |
| QA checks                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1696992528/164a81877b096e4df72869861310e909b2103d32/destination-redshift/0.6.10/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:164a81877b096e4df72869861310e909b2103d32)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-redshift test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-snowflake test report (commit [`164a81877b`](https://github.com/airbytehq/airbyte/pull/31194/commits/164a81877b096e4df72869861310e909b2103d32)) - ❌

⏲️  Total pipeline duration: 01mn09s 

| Step                                        | Result   |
|:--------------------------------------------|:---------|
| Build connector tar                         | ❌        |
| Validate metadata for destination-snowflake | ✅        |
| Connector version semver check              | ✅        |
| Connector version increment check           | ✅        |
| QA checks                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1696992528/164a81877b096e4df72869861310e909b2103d32/destination-snowflake/3.2.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:164a81877b096e4df72869861310e909b2103d32)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-snowflake test report (commit [`1a7fad8213`](https://github.com/airbytehq/airbyte/pull/31194/commits/1a7fad82139b101a237f3fdc1d2ba8f2058217c2)) - ❌

⏲️  Total pipeline duration: 54.47s 

| Step                                        | Result   |
|:--------------------------------------------|:---------|
| Build connector tar                         | ❌        |
| Validate metadata for destination-snowflake | ✅        |
| Connector version semver check              | ✅        |
| Connector version increment check           | ✅        |
| QA checks                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1697000415/1a7fad82139b101a237f3fdc1d2ba8f2058217c2/destination-snowflake/3.2.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1a7fad82139b101a237f3fdc1d2ba8f2058217c2)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`1a7fad8213`](https://github.com/airbytehq/airbyte/pull/31194/commits/1a7fad82139b101a237f3fdc1d2ba8f2058217c2)) - ❌

⏲️  Total pipeline duration: 01mn20s 

| Step                                       | Result   |
|:-------------------------------------------|:---------|
| Build connector tar                        | ❌        |
| Validate metadata for destination-bigquery | ✅        |
| Connector version semver check             | ✅        |
| Connector version increment check          | ✅        |
| QA checks                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1697000415/1a7fad82139b101a237f3fdc1d2ba8f2058217c2/destination-bigquery/2.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1a7fad82139b101a237f3fdc1d2ba8f2058217c2)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-redshift test report (commit [`1a7fad8213`](https://github.com/airbytehq/airbyte/pull/31194/commits/1a7fad82139b101a237f3fdc1d2ba8f2058217c2)) - ❌

⏲️  Total pipeline duration: 22.00s 

| Step                                       | Result   |
|:-------------------------------------------|:---------|
| Build connector tar                        | ❌        |
| Validate metadata for destination-redshift | ✅        |
| Connector version semver check             | ✅        |
| QA checks                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1697000415/1a7fad82139b101a237f3fdc1d2ba8f2058217c2/destination-redshift/0.6.10/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:1a7fad82139b101a237f3fdc1d2ba8f2058217c2)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-redshift test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-snowflake test report (commit [`7566b0f122`](https://github.com/airbytehq/airbyte/pull/31194/commits/7566b0f122f0944d28daea43ec42517554177b2d)) - ❌

⏲️  Total pipeline duration: 46.60s 

| Step                                        | Result   |
|:--------------------------------------------|:---------|
| Build connector tar                         | ❌        |
| Validate metadata for destination-snowflake | ✅        |
| Connector version semver check              | ✅        |
| Connector version increment check           | ✅        |
| QA checks                                   | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1697001322/7566b0f122f0944d28daea43ec42517554177b2d/destination-snowflake/3.2.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:7566b0f122f0944d28daea43ec42517554177b2d)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-redshift test report (commit [`7566b0f122`](https://github.com/airbytehq/airbyte/pull/31194/commits/7566b0f122f0944d28daea43ec42517554177b2d)) - ❌

⏲️  Total pipeline duration: 21.86s 

| Step                                       | Result   |
|:-------------------------------------------|:---------|
| Build connector tar                        | ❌        |
| Validate metadata for destination-redshift | ✅        |
| Connector version semver check             | ✅        |
| QA checks                                  | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1697001322/7566b0f122f0944d28daea43ec42517554177b2d/destination-redshift/0.6.10/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:7566b0f122f0944d28daea43ec42517554177b2d)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-redshift test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`f52bb50d84`](https://github.com/airbytehq/airbyte/pull/31194/commits/f52bb50d84f24edb7ff42a8864c0a3ff86a48f01)) - ✅

⏲️  Total pipeline duration: 08mn38s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Build connector tar                                                  | ✅        |
| Build destination-bigquery docker image for platform(s) linux/x86_64 | ✅        |
| Java Connector Unit Tests                                            | ✅        |
| Java Connector Integration Tests                                     | ✅        |
| Validate metadata for destination-bigquery                           | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1697041782/f52bb50d84f24edb7ff42a8864c0a3ff86a48f01/destination-bigquery/2.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:f52bb50d84f24edb7ff42a8864c0a3ff86a48f01)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-snowflake test report (commit [`f52bb50d84`](https://github.com/airbytehq/airbyte/pull/31194/commits/f52bb50d84f24edb7ff42a8864c0a3ff86a48f01)) - ✅

⏲️  Total pipeline duration: 10mn13s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Build connector tar                                                   | ✅        |
| Build destination-snowflake docker image for platform(s) linux/x86_64 | ✅        |
| Java Connector Unit Tests                                             | ✅        |
| Java Connector Integration Tests                                      | ✅        |
| Validate metadata for destination-snowflake                           | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1697041782/f52bb50d84f24edb7ff42a8864c0a3ff86a48f01/destination-snowflake/3.2.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:f52bb50d84f24edb7ff42a8864c0a3ff86a48f01)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-redshift test report (commit [`dc98428471`](https://github.com/airbytehq/airbyte/pull/31194/commits/dc984284715c9f04ae3ae5fbbb586e70e429b8fc)) - ✅

⏲️  Total pipeline duration: 20mn38s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Build connector tar                                                  | ✅        |
| Build destination-redshift docker image for platform(s) linux/x86_64 | ✅        |
| Java Connector Unit Tests                                            | ✅        |
| Build airbyte/normalization-redshift:dev                             | ✅        |
| Java Connector Integration Tests                                     | ✅        |
| Validate metadata for destination-redshift                           | ✅        |
| Connector version semver check                                       | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1697043148/dc984284715c9f04ae3ae5fbbb586e70e429b8fc/destination-redshift/0.6.9/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:dc984284715c9f04ae3ae5fbbb586e70e429b8fc)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-redshift test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-bigquery test report (commit [`dc98428471`](https://github.com/airbytehq/airbyte/pull/31194/commits/dc984284715c9f04ae3ae5fbbb586e70e429b8fc)) - ✅

⏲️  Total pipeline duration: 08mn00s 

| Step                                                                 | Result   |
|:---------------------------------------------------------------------|:---------|
| Build connector tar                                                  | ✅        |
| Build destination-bigquery docker image for platform(s) linux/x86_64 | ✅        |
| Java Connector Unit Tests                                            | ✅        |
| Java Connector Integration Tests                                     | ✅        |
| Validate metadata for destination-bigquery                           | ✅        |
| Connector version semver check                                       | ✅        |
| Connector version increment check                                    | ✅        |
| QA checks                                                            | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1697043148/dc984284715c9f04ae3ae5fbbb586e70e429b8fc/destination-bigquery/2.1.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:dc984284715c9f04ae3ae5fbbb586e70e429b8fc)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by airbyte-oss-build-runner** [comment]
##  destination-snowflake test report (commit [`dc98428471`](https://github.com/airbytehq/airbyte/pull/31194/commits/dc984284715c9f04ae3ae5fbbb586e70e429b8fc)) - ✅

⏲️  Total pipeline duration: 10mn21s 

| Step                                                                  | Result   |
|:----------------------------------------------------------------------|:---------|
| Build connector tar                                                   | ✅        |
| Build destination-snowflake docker image for platform(s) linux/x86_64 | ✅        |
| Java Connector Unit Tests                                             | ✅        |
| Java Connector Integration Tests                                      | ✅        |
| Validate metadata for destination-snowflake                           | ✅        |
| Connector version semver check                                        | ✅        |
| Connector version increment check                                     | ✅        |
| QA checks                                                             | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/rbroughan_clean-up-empty-buffer-memory/1697043148/dc984284715c9f04ae3ae5fbbb586e70e429b8fc/destination-snowflake/3.2.2/output.html)

☁️ [View runs for commit in Dagger Cloud](https://alpha.dagger.cloud/changeByPipelines?filter=dagger.io/git.ref:dc984284715c9f04ae3ae5fbbb586e70e429b8fc)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Review by benmoriceau — APPROVED** [review]

**Pull request merged**